clang/include/clang/AST/Stmt.h

```diff
@@ -3326,9 +3325,11 @@ class GCCAsmStmt : public AsmStmt {
 
   //===--- Asm String Analysis ---===//
 
-  const StringLiteral *getAsmString() const { return AsmStr; }
-  StringLiteral *getAsmString() { return AsmStr; }
-  void setAsmString(StringLiteral *E) { AsmStr = E; }
+  const Expr *getAsmStringExpr() const { return AsmStr; }
+  Expr *getAsmStringExpr() { return AsmStr; }
+  void setAsmString(Expr *E) { AsmStr = E; }
```

Comment:
this should either be `setAsmStringExpr` or the above be `getAsmString`.  I don't really car which.  THOUGH we are double-using `getAsmString` on 3332, so I suggest the former.  I can't find a way to set the Constraint or Clobbers though.

Reply:
not sure I get this comment?

Reply:
I see now... that one I think just leave alone, its a whole thing.  But `setAsmStringExpr` to differentiate from `getAsmString` and `getAsmStringExpr` seems sensible.